pycontrails/core/fleet.py

```diff
@@ -174,10 +178,11 @@ def from_seq(
             instances in ``seq``. The fuel type is taken from the first :class:`Flight`
             in ``seq``.
         """
-        if copy:
-            seq = tuple(fl.copy() for fl in seq)
-        elif not isinstance(seq, (list, tuple)):
-            seq = tuple(seq)
+
+        def _maybe_copy(fl: Flight) -> Flight:
+            return fl.copy() if copy else fl
+
+        seq = tuple(_maybe_copy(fl) for fl in seq if fl)  # filter out empty flights
```

Comment:
Is it worth emitting a warning if some empty flights are filtered out? I'm not sure why you would intentionally pass in an empty flight, so this might be a signal that something has gone wrong in an earlier preprocessing step.

Reply:
@thabbott Yeah, I thought about this exact question, and I wasn't sure! I'll completely defer to you here. You encountered this issue in a notebook in which you called `Fleet.resample_and_fill`. Would you have been happier with a warning? I think a warning makes sense if there's something you could have done differently -- should you have previously removed the short flights?

Reply:
e2871f6db3fb267169f425d9ff7d4ac4cee88292